### PR TITLE
Make Torch installation explicit

### PR DESCRIPTION
**Pull request by ternaus**
## What changed

- Removes Torch and TorchVision from the published base dependency set. Importing `albumentations` still requires the Torch build selected by the user; a missing Torch installation now gives a direct installation command.
- Routes the dedicated CI and benchmark dependency groups through the PyTorch CPU index, so ordinary CI and documentation/package-only installs do not resolve Torch or CUDA packages.
- Raises Albucore to `0.2.13` and NumKong to `>=7.8.0`; conda metadata and lower-bound CI are aligned.
- Removes the local `apply_multichannel_lut` helper. Equalize and AutoContrast build the final `(256, 1, C)` LUT directly and delegate application to `albucore.apply_uint8_lut`; ToneCurve uses the native `(256, C)` result with a zero-copy axis insertion.
- Updates the dev lock group and declared floors: pre-commit 4.6.2, Ruff 0.16.2, Pyrefly 1.2.0, Zizmor 1.29.0, and related tooling.

## Why

Installing the package for documentation, website builds, or non-runtime checks must not force a Torch/CUDA resolver decision. Users select the CPU, CUDA, or MPS Torch distribution appropriate to their environment; CI intentionally uses only CPU Torch where it is required.

## Validation

- `pre-commit autoupdate` (all configured external hook revisions already current)
- `uv run pytest -q -n auto -m "not slow"` — 13,363 passed, 43 skipped, 9 xfailed, 3 xpassed
- `uv run python tools/quality_gate.py fast`
- `uv run python -m tools.ci_matrix check` and `uv run python -m tools.ci_shard check`
- `uv run pytest -q tests/test_ci_plan.py tests/test_ci_gate.py tests/test_ci_shard.py tests/test_pr_workflow.py` — 57 passed
- `uv run zizmor --format=plain --min-severity=medium --min-confidence=medium .github` — no findings
- Built fresh wheel/sdist; `verify_legal_integrity.py --artifacts` and `twine check` passed.

## LUT benchmark

Exact old/new output equality was checked for direct and Compose routes across uint8 256, 512, and 1024 pixels with C=1/3/5. The direct and Compose matrices stayed within the 5% regression threshold; the change removes temporary `np.stack`/transpose/contiguous-copy allocations.

## Summary by Sourcery

Make PyTorch a soft-required runtime dependency while keeping it out of the base install, and align CI, packaging, and documentation with the new Torch and LUT handling contracts.

New Features:
- Add explicit runtime checks that OpenCV and PyTorch are installed, with a guided PyTorch installation message when missing.
- Document that users must install a CPU, CUDA, or MPS PyTorch build separately and provide concrete installation commands and guidance.
- Introduce tests that enforce the optional/Torch import contracts and five-dimensional input rejection for selected color operations.

Bug Fixes:
- Ensure multichannel equalize, autocontrast, and tone-curve operations preserve the channel dimension and behave consistently across shapes.
- Tighten color functional APIs to reject unsupported five-dimensional inputs instead of silently reshaping them.

Enhancements:
- Replace the local multichannel LUT helper with direct use of albucore.apply_uint8_lut to avoid extra allocations and simplify LUT handling.
- Constrain linear RGB transforms and grayscale conversions to 3D/4D layouts, with clearer error messages for unsupported shapes.
- Raise Albucore and NumKong minimum versions and refresh several tooling floors (pre-commit, Ruff, Pyrefly, Zizmor, types stubs, and related libs).

Build:
- Update pyproject and conda metadata to drop Torch from base dependencies, bump Albucore and NumKong floors, and ensure Torch is only a test dependency in Conda.
- Configure uv to route Torch installation through the explicit pytorch-cpu index and add checks that CI dependency groups only install Torch where expected.

CI:
- Adjust CI dependency groups so only PyTorch and benchmark profiles install Torch, and add a validator enforcing Torch/TorchVision placement and the pytorch-cpu index routing.
- Simplify PyTorch jobs to rely on the ci-pytorch dependency group and verify CPU-only Torch profiles instead of manually installing Torch/TorchVision.
- Extend smoke tests to confirm base wheels do not install Torch, that importing without Torch fails with the documented guidance, and that adding Torch via the CPU index restores imports.
- Update PR workflow tests and policy docs to reflect the new CPU-only Torch profile and separation of Torch from general compatibility jobs.

Documentation:
- Revise README and maintenance/design docs to describe Torch as a soft-required runtime dependency, explain installation flows, and clarify CI/support expectations around Torch builds and extras.

Tests:
- Remove torchvision-based color parity tests and add new tests for channel-dimension preservation, five-dimensional input rejection, and the optional Torch import behavior.
- Update CI matrix, PR workflow, and optional extras smoke tests to align with the new Torch handling and dependency floors.

### CODE REVIEW

File: albumentations/__init__.py [hunk 1]
```diff
@@ -20,6 +24,22 @@
     )
     raise ImportError(msg) from e
 
+try:
+    import torch  # noqa: F401
```
[the remark this answers is not in the document]
Comment (ternaus) [reply albumentations/__init__.py:28]: Fixed in the current head (`570209a`). Every PR job that imports `albumentations` now explicitly uses `runtime-profile: torch-cpu`; the shared setup action installs `ci-torch-cpu`, and `tools/ci_matrix.py` plus `tests/test_ci_runtime_profiles.py` enforce that contract. The full cross-platform CI matrix is green.
